src/gurt/telemetry.c

```diff
@@ -2417,37 +2433,73 @@ d_tm_get_shared_memory(int srv_idx)
 			"%s\n", strerror(errno));
 		return NULL;
 	}
-	return addr;
+
+	rc = alloc_ctx(&new_ctx, addr);
```

Comment:
rc is a redundant variable here, can just test if (alloc_ctx(...) != 0)

Reply:
Good point, can fix in a follow-on